src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -160,6 +160,10 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
                     // input value among many constant tests.
                     convertedExpression = operand;
                 }
+                else if (conversion.ConversionKind == ConversionKind.NoConversion && convertedExpression.Type?.IsErrorType() == true)
+                {
+                    convertedExpression = operand;
```

Comment:
When do we hit this case? And by dropping the original `convertedExpression` node, is it possible there are no nodes with `HasErrors` as a result? (Is that an issue?)